### PR TITLE
Update README FAQ with latest plan

### PR DESCRIPTION
**Pull request by shadowspawn**
Update the README FAQ with changes from #21 and #22, short option group plans from #2, and some updates to speculative or slightly incorrect comments.

I also updated the `--` QA to match the current behaviour. (With which I agree.)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,13 +146,13 @@ const { flags, values, positionals } = parseArgs(argv, options);
 ### F.A.Qs
 
 - Is `cmd --foo=bar baz` the same as `cmd baz --foo=bar`?
-  - Yes, if `withValue: ['foo']`, otherwise no
+  - Yes
```
Comment (Eomm) [inline README.md:149]: This is the only answer with the drop cap

```suggestion
  - yes
```

Comment (shadowspawn) [reply README.md]: Fixed, thanks

**Review by Eomm — APPROVED** [review]

**Comment by bcoe** [comment]
what if we make an effort to bring the discussion in #38 to a conclusion, and then update the examples in the README as needed?

**Comment by shadowspawn** [comment]
This was intended to be fairly safe catch up with multiple small fixes. Failed. Closing PR to avoid discussion here which I don't feel will be best place or very focused.

Will submit some of the hopefully no-brainer changes individually so easy to yes/no quickly.

**Pull request closed without merging**